### ISSUE TITLE
patch request with body fails on Android

### ISSUE THREAD

**Issue by markbrown4** [facebook/react-native#7137]
I get the following error when making a 'patch' request with fetch on android, works as expected on ios.

> method patch must not have a request body
> Request.java:253

```js
fetch('/api', {
  method: 'patch',
  headers: {
    'Accept': 'application/json',
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({
    booking: {
      booking_resource_id: 3,
      checked_in_at: 1461304730734,
      starts_at: 1461304800000
    }
  })
})
```

**Comment by grabbou** [comment]
Have you tried using `PATCH` instead of `patch` ?

**Comment by grabbou** [comment]
I cannot reproduce on master.

```js
fetch('http://google.com/api', {
      method: 'patch',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        booking: {
          booking_resource_id: 3,
          checked_in_at: 1461304730734,
          starts_at: 1461304800000
        }
      })
    })
    .then(res => {
      console.log(res.status);  // 404
    });
```
PS. Changing to `GET` raises an error, so that part works.

**Comment by markbrown4** [comment]
You're earlier comment was on the money, it works with `PATCH` but not `patch`.

Testing on a "Google Nexus 10 - 5.0.0 API 21" emulator in GenyMotion on a mac.

**Comment by grabbou** [comment]
Yeah, that's what I thought but then I found it's already uppercased in master. Looks like fixed in #6956
